services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -157,6 +156,7 @@ public void configure(Binder binder)
             binder.bind(DataSegmentArchiver.class).to(OmniDataSegmentArchiver.class).in(LazySingleton.class);
 
             binder.bind(ExecutorLifecycle.class).in(ManageLifecycle.class);
```

Comment:
This will tell the primary Lifecycle to manage the ExecutorLifecycle, so yes, this should still be needed.